File: solutions/zig/src/solutions.zig
const std = @import("std");
const year2022 = @import("year2022/year2022.zig");

const Solution = *const fn (input: []const u8, allocator: std.mem.Allocator) anyerror!struct { []u8, []u8 };

pub const map = std.ComptimeStringMap(Solution, .{
    .{ "2022/01", year2022.day01.solution },
    .{ "2022/02", year2022.day02.solution },
});

File: solutions/haskell/src/Solutions.hs
module Solutions (solutions) where

import qualified Data.Map as Map

import qualified Year2022.Day01
import qualified Year2022.Day02

solutions :: Map.Map String (String -> (String, String))
solutions =
    Map.fromList
        [ ("2022/01", Year2022.Day01.solution)
        , ("2022/02", Year2022.Day02.solution)
        ]
